<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>803</width>
    <height>563</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Mesh_Computational_Geometry</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="sizePolicy">
    <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <widget class="GLDisplayWidget" name="widget" native="true">
    <property name="geometry">
     <rect>
      <x>150</x>
      <y>10</y>
      <width>601</width>
      <height>361</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>520</y>
      <width>80</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Exit</string>
    </property>
   </widget>
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>120</width>
      <height>101</height>
     </rect>
    </property>
    <property name="title">
     <string>Display</string>
    </property>
    <widget class="QWidget" name="layoutWidget">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>30</y>
       <width>100</width>
       <height>62</height>
      </rect>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QRadioButton" name="pushButton_2">
        <property name="text">
         <string>Wireframe</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QRadioButton" name="pushButton_3">
        <property name="text">
         <string>Plain</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </widget>
   <widget class="QPushButton" name="randFlip">
    <property name="geometry">
     <rect>
      <x>9</x>
      <y>130</y>
      <width>121</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Random Flip</string>
    </property>
   </widget>
   <widget class="QPushButton" name="randomFH">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>160</y>
      <width>121</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Random Face
Highlight</string>
    </property>
   </widget>
   <widget class="QPushButton" name="unmark">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>210</y>
      <width>101</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Remove
Highlights</string>
    </property>
    <property name="autoDefault">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QGroupBox" name="groupBox_2">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>260</y>
      <width>120</width>
      <height>111</height>
     </rect>
    </property>
    <property name="title">
     <string>Face Circulator</string>
    </property>
    <widget class="QLineEdit" name="axisIndex">
     <property name="geometry">
      <rect>
       <x>60</x>
       <y>20</y>
       <width>51</width>
       <height>16</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="label">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>45</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Vertex</string>
     </property>
    </widget>
    <widget class="QPushButton" name="nextFace">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>50</y>
       <width>71</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>-&gt;</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_3">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>390</y>
      <width>120</width>
      <height>121</height>
     </rect>
    </property>
    <property name="title">
     <string>Insertion</string>
    </property>
    <widget class="QLineEdit" name="xInsert">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>21</width>
       <height>20</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="yInsert">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>20</y>
       <width>21</width>
       <height>20</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="zInsert">
     <property name="geometry">
      <rect>
       <x>90</x>
       <y>20</y>
       <width>21</width>
       <height>20</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="x">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>40</y>
       <width>16</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>   x</string>
     </property>
    </widget>
    <widget class="QLabel" name="y">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>40</y>
       <width>16</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>   y</string>
     </property>
    </widget>
    <widget class="QLabel" name="z">
     <property name="geometry">
      <rect>
       <x>90</x>
       <y>40</y>
       <width>16</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>   z</string>
     </property>
    </widget>
    <widget class="QPushButton" name="naiveInsert">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>60</y>
       <width>80</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Naive Insert</string>
     </property>
    </widget>
   </widget>
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>170</x>
      <y>410</y>
      <width>211</width>
      <height>121</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>1</number>
    </property>
    <widget class="QWidget" name="Delaunay">
     <attribute name="title">
      <string>Delaunay</string>
     </attribute>
    </widget>
    <widget class="QWidget" name="Voronoi">
     <attribute name="title">
      <string>Voronoi</string>
     </attribute>
     <widget class="QPushButton" name="computeVoronoi">
      <property name="geometry">
       <rect>
        <x>60</x>
        <y>60</y>
        <width>91</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Compute Voronoi</string>
      </property>
     </widget>
     <widget class="QCheckBox" name="showVoronoi">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>91</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Show Voronoi</string>
      </property>
     </widget>
    </widget>
   </widget>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>GLDisplayWidget</class>
   <extends>QWidget</extends>
   <header>gldisplaywidget.h</header>
   <container>1</container>
   <slots>
    <slot>onWireframe()</slot>
    <slot>onPlain()</slot>
    <slot>randomFlip()</slot>
    <slot>randomFHighlight()</slot>
    <slot>unmark()</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>pushButton</sender>
   <signal>released()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>63</x>
     <y>472</y>
    </hint>
    <hint type="destinationlabel">
     <x>120</x>
     <y>467</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>randFlip</sender>
   <signal>released()</signal>
   <receiver>widget</receiver>
   <slot>randomFlip()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>88</x>
     <y>154</y>
    </hint>
    <hint type="destinationlabel">
     <x>294</x>
     <y>151</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>randomFH</sender>
   <signal>released()</signal>
   <receiver>widget</receiver>
   <slot>randomFHighlight()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>107</x>
     <y>191</y>
    </hint>
    <hint type="destinationlabel">
     <x>175</x>
     <y>194</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
